### PR TITLE
validation, error handling, redirect, http-post

### PR DESCRIPTION
**Pull request by jakzda**
https://trello.com/c/Jr07F0D9/5-fe-03-create-new-recipe-m

### CODE REVIEW

File: src/app/recipes/recipe-form/recipe-form.component.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        Validators.required
+      ])
+    })
+  }
+
+  addRecipe(){
+    const recipe: PostRecipe = this.recipeForm.value;
+    this.recipeService.postRecipe(recipe).pipe(
+      catchError( err=> {
+        console.log('Handling error locally and rethrowing it...', err);
```
Comment (Isterius) [inline src/app/recipes/recipe-form/recipe-form.component.ts:38]: not needed

File: src/app/recipes/recipe-form/recipe-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        return throwError(err);
+      })
+    )
+    .subscribe(
+      (res:any) => {
+        console.log('HTTP response', res);
```
Comment (Isterius) [inline src/app/recipes/recipe-form/recipe-form.component.ts:44]: not needed

File: src/app/recipes/recipe.service.ts [hunk 3]
```diff
@@ -41,6 +41,9 @@ export class RecipeService {
     }))
   }
 
+  postRecipe(recipe: PostRecipe): Observable<PostRecipe> {
+    return this.http.post<PostRecipe>('https://cookbook-be-team-4.herokuapp.com/api/v1/recipes', recipe) }
```
Comment (Isterius) [inline src/app/recipes/recipe.service.ts:45]: are you sure this return type is proper and backend is returning `Recipe` without id? 👀

**Review by Isterius — CHANGES_REQUESTED** [review]

File: src/app/recipes/recipe.service.ts (hunk 3 above)
Comment (jakzda) [reply src/app/recipes/recipe.service.ts:45]: yes i'm sure, I even asked backend boys, Michał and Daniel before I have written this code.

Comment (Isterius) [reply src/app/recipes/recipe.service.ts:45]: this is what backend returns and it's not assignable you your `PostRecipe` interface
image

**Review by Isterius — CHANGES_REQUESTED** [review]

**Review by Isterius — APPROVED** [review]

**Pull request merged**